src/iproject.ts

```diff
@@ -44,42 +104,87 @@ export class IProject {
     this.environmentValues = {};
   }
 
+  /**
+   * Get the project's name. *Note* that the project's name and associated
+   * workspace folder name are the same.
+   * 
+   * @returns Project name.
+   */
   public getName(): string {
     return this.name;
   }
 
+  /**
+   * Get the uri of a project file. *Note* that the `directory` parameter should
+   * be passed in when the desired project file is a directory level `.ibmi.json`
+   * that is not in the root directory.
+   * 
+   * @param type A type of project file.
+   * @param directory The directory containing the project file.
+   * @returns A uri of the desired resource.
+   */
   public getProjectFileUri(type: ProjectFileType, directory?: Uri): Uri {
     const logDirectory = (type === 'joblog.json' || type === 'output.log') ? `.logs` : ``;
 
     return Uri.file(path.join(directory ? directory.fsPath : this.workspaceFolder.uri.fsPath, logDirectory, type));
   }
 
+  /**
+   * Check if a project file exists. *Note* that the `directory` parameter should
+   * be passed in when the desired project file is a directory level `.ibmi.json`
+   * that is not in the root directory.
+   * 
+   * @param type A type of project file.
+   * @param directory The directory containing the project file.
```

Comment:
is this a relative to project root directory or absolute path?

Reply:
Since we are dealing with a `uri` here, it will always be absolute.